### PR TITLE
Add PDS Base Image to FindSecurityBugs Docker Compose files

### LINKED ISSUE #1860: FindSecurityBugs+PDS still using Debian as Base Image
# Situation
`FindSecurityBugs-Debian.dockerfile` was adjusted to work with `PDS-Debian.dockerfile` as `BASE_IMAGE` (#1723) but Docker Compose files for FindSecurityBugs still considers `debian:11-slim` as `BASE_IMAGE`.  

This results in the following error:  
```bash
Using existing environment file: .env.
Starting single container with external network.
[+] Building 29.2s (12/13)
 => [internal] load build definition from FindSecurityBugs-Debian.dockerfile                                                                                                                                                                  0.0s
 => => transferring dockerfile: 2.44kB                                                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                             0.0s
 => => transferring context: 2B                                                                                                                                                                                                               0.0s
 => [internal] load metadata for docker.io/library/debian:11-slim                                                                                                                                                                             0.4s
 => CACHED [ 1/10] FROM docker.io/library/debian:11-slim@sha256:8c60e19bd285355325cf3e1d39223317ec5847346726c5ae88a037c01322b426                                                                                                              0.0s
 => [internal] load build context                                                                                                                                                                                                             0.0s
 => => transferring context: 82.84kB                                                                                                                                                                                                          0.0s
 => [ 2/10] RUN export DEBIAN_FRONTEND=noninteractive &&     apt-get update &&     apt-get --assume-yes upgrade  &&     apt-get --assume-yes install dos2unix unzip wget openjdk-17-jre-headless libxml2-utils &&     apt-get --assume-yes   24.1s
 => [ 3/10] RUN cd "$DOWNLOAD_FOLDER" &&     wget --no-verbose "https://github.com/find-sec-bugs/find-sec-bugs/releases/download/version-1.12.0/findsecbugs-cli-1.12.0.zip"  &&     echo "a50bd4741a68c6886bbc03d20da9ded44bce4dd7d0d2eee19c  4.2s
 => [ 4/10] COPY pds-config.json //pds-config.json                                                                                                                                                                                            0.0s
 => [ 5/10] COPY findsecbugs_sechub.sh /findsecbugs_sechub.sh                                                                                                                                                                                 0.0s
 => [ 6/10] RUN chmod +x $TOOL_FOLDER/findsecbugs_sechub.sh                                                                                                                                                                                   0.1s
 => [ 7/10] COPY scripts                                                                                                                                                                                                                      0.0s
 => ERROR [ 8/10] RUN chmod --recursive +x $SCRIPT_FOLDER                                                                                                                                                                                     0.2s
------
 > [ 8/10] RUN chmod --recursive +x $SCRIPT_FOLDER:
#0 0.150 chmod: missing operand after '+x'
#0 0.150 Try 'chmod --help' for more information.
``` 

due to lacking `ARG`s and `ENV` variables in Dockerfile. 

# Wanted
Containers for PDS-FindSecurityBugs should start with `PDS-Debian.dockerfile` as `BASE_IMAGE`.

# Solution
Set argument `BASE_IMAGE` to `${BASE_IMAGE}` in following Docker Compose files:  
* docker-compose_pds_findsecuritybugs.yaml 
* docker-compose_pds_findsecuritybugs_cluster.yaml  
* docker-compose_pds_findsecuritybugs_cluster_object_storage.yaml  
* docker-compose_pds_findsecuritybugs_external-network.yaml

### PR DESCRIPTION
**Pull request by ddzikon**
This PR adds `${BASE_IMAGE}` to following Docker Compose files for FindSecurityBugs:  
* docker-compose_pds_findsecuritybugs.yaml
* docker-compose_pds_findsecuritybugs_cluster.yaml
* docker-compose_pds_findsecuritybugs_cluster_object_storage.yaml
* docker-compose_pds_findsecuritybugs_external-network.yaml

Value for `BASE_IMAGE` can be set in files containing initial values:
* `env-initial`
* `env-cluster-initial`

closes #1860

### CODE REVIEW

**Comment by ddzikon** [comment]
There is a slight difference between versions:
* `PDS_VERSION=0.31.0`
* `BASE_IMAGE="ghcr.io/mercedes-benz/sechub/pds-base:v0.32.1"`

I see that most images using `pds-base` consider version `0.32.1`.

Is it ok to have it so or should it be addressed?

**Comment by Jeeppler** [comment]
@ddzikon thanks for your contribution.

The local docker-compose files for the FindSecurityBugs solution are indeed not working.

We introduced the PDS base image, after we figured out that we can: a) push image to ghcr.io and that b) images can be hosted publicly on ghcr.io (ghcr.io is the second attempt of GitHub to create a container registry).

Variables like the `PDS_VERSION` are already set in the PDS-Base image and are therefore redundant. All variables set in the base image are inherited by all PDS-Solutions using the base image. So, please have a look at the Dockerfile of the PDS-Base image: https://github.com/mercedes-benz/sechub/blob/develop/sechub-pds-solutions/pds-base/docker/PDS-Debian.dockerfile#L122. That means the `PDS_VERSION` variable can be removed, it is enough to use the `BASE_IMAGE` going forward.

Another simplification which we did some time ago, was to simply reference the standard environment files. As they are all the same. Have a look at the file from GoSec: https://github.com/mercedes-benz/sechub/blob/develop/sechub-pds-solutions/gosec/01-start-single-docker-compose.sh:

~~~
ENVIRONMENT_FILES_FOLDER="../shared/environment"
ENVIRONMENT_FILE=".env-single"

# Only variables from .env can be used in the Docker-Compose file
# all other variables are only available in the container
setup_environment_file ".env" "env"
setup_environment_file "$ENVIRONMENT_FILE" "$ENVIRONMENT_FILES_FOLDER/env-base"
~~~

Like in GoSec, there should only be a single "env" file with the base image variable in it. The 01… scripts copy the values from the shared files and create an `.env` and a `.env-single` etc. file.

Please incorporate those changes into your pull-request.

File: sechub-pds-solutions/findsecuritybugs/docker-compose_pds_findsecuritybugs.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ services:
     pds-findsecuritybugs:
         build:
             args:
-                - BASE_IMAGE=debian:11-slim
+                - BASE_IMAGE=${BASE_IMAGE}
                 - PDS_VERSION=${PDS_VERSION}
```
Comment (Jeeppler) [inline sechub-pds-solutions/findsecuritybugs/docker-compose_pds_findsecuritybugs.yaml:9]: Please remove the `PDS_VERSION` variable: [#1862#issuecomment-1382224019](https://github.com/mercedes-benz/sechub/pull/1862#issuecomment-1382224019)

File: sechub-pds-solutions/findsecuritybugs/env-cluster-initial [hunk 2]
```diff
@@ -19,4 +19,5 @@ S3_ACCESSKEY=user
 S3_SECRETKEY=Why$5forThat?
```
Comment (Jeeppler) [inline sechub-pds-solutions/findsecuritybugs/env-cluster-initial:19]: This file should be removed. For more information have a look at: https://github.com/mercedes-benz/sechub/pull/1862#issuecomment-1382224019.

File: sechub-pds-solutions/findsecuritybugs/env-initial [hunk 3]
```diff
@@ -7,4 +7,5 @@ JAVA_ENABLE_DEBUG=false
 KEEP_CONTAINER_ALIVE_AFTER_PDS_CRASHED=false
```
Comment (Jeeppler) [inline sechub-pds-solutions/findsecuritybugs/env-initial:7]: This file should be removed. For more information have a look at: https://github.com/mercedes-benz/sechub/pull/1862#issuecomment-1382224019.

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Thanks for your contribution.

I found a number of issues and described them in: https://github.com/mercedes-benz/sechub/pull/1862#issuecomment-1382224019.

Feel free to ask more questions if you have any.

**Comment by ddzikon** [comment]
@Jeeppler thank you for the remarks!

I removed `PDS_VERSION` variable and made scripts use shared `env` files and helper scripts, like it is done in GoSec.  

File `env-cluster-inital` for FindSecurityBugs contained also config variables for database connection.  
Should FindSecurityBugs have separate `env-database` file like there is one in GoSec?  
At the moment, i removed use of this file from scripts `50-start-multiple-docker-compose.sh` and `51-start-multiple-object-storage-docker-compose.sh`.

Please let me know, if there are more adjustments to be done.

File: sechub-pds-solutions/findsecuritybugs/docker-compose_pds_findsecuritybugs_external-network.yaml [hunk 4]
```diff
@@ -6,13 +6,13 @@ services:
...
             args:
                 - BASE_IMAGE=${BASE_IMAGE}
-                - PDS_VERSION=${PDS_VERSION}
             context: docker/
             dockerfile: FindSecurityBugs-Debian.dockerfile
         container_name: pds-findsecuritybugs
         hostname: pds-findsecuritybugs
         env_file:
             - .env
+            - .emv-single
```
Comment (Jeeppler) [inline sechub-pds-solutions/findsecuritybugs/docker-compose_pds_findsecuritybugs_external-network.yaml:15]:
```suggestion
            - .env-single
```

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comment.

**Comment by Jeeppler** [comment]
> Should FindSecurityBugs have separate env-database file like there is one in GoSec?

Yes, that file is important. Each Tool+PDS combination needs to have it's own database.

File: sechub-pds-solutions/findsecuritybugs/docker-compose_pds_findsecuritybugs_external-network.yaml (hunk 4 above)
Comment (ddzikon) [reply sechub-pds-solutions/findsecuritybugs/docker-compose_pds_findsecuritybugs_external-network.yaml]: Good catch, sorry for that typo 😅

**Comment by ddzikon** [comment]
> Yes, that file is important. Each Tool+PDS combination needs to have it's own database.

I recreated this file with configuration based on the contents of old `env-cluster-initial` file and adequate files for GoSec.

**Review by Jeeppler — APPROVED** [review]
Looks good 🥇 . Thanks for your contribution.

**Pull request merged**